### PR TITLE
BD-10 feat: add button component

### PR DESCRIPTION
**Pull request by pRomKol**

### CODE REVIEW

File: src/components/button/Button.tsx [hunk 1]
```diff
@@ -1,60 +1,80 @@
...
+          'bg-transparent text-accent-500 border border-accent-500',
+          ' enabled:hover:border-accent-100 enabled:hover:text-accent-100 active:text-accent-700 active:border-accent-700',
+        ],
+        primary: [
+          'bg-accent-500',
+          ' enabled:hover:bg-accent-100 active:bg-accent-700 active: text-light-500',
+        ],
+        secondary: [
+          'bg-dark-300',
+          ' enabled:hover:bg-dark-100 active:bg-dark-500  focus:outline-accent-500',
```
Comment (SlavaMryasov) [inline src/components/button/Button.tsx:53]: focus doesn't work everywhere

File: src/components/button/Button.tsx [hunk 2]
```diff
@@ -1,60 +1,80 @@
...
-  const primaryClasses =
-    'bg-blue-500 text-white hover:bg-blue-700 focus:ring-blue-500'
-  const secondaryClasses =
-    'bg-gray-200 text-black hover:bg-gray-400 focus:ring-gray-500'
+const buttonVariants = cva(
+  ['text-white text-base font-semibold py-1.5 px-6 rounded-sm'],
+  {
+    compoundVariants: [
+      {
+        class: 'text-accent-900 opacity-50 border-accent-700',
```
Comment (SlavaMryasov) [inline src/components/button/Button.tsx:13]: where did you get opacity, remove!

**Review by SlavaMryasov — CHANGES_REQUESTED** [review]
make the components behave as in the design

File: src/components/button/Button.tsx [hunk 3]
```diff
@@ -1,60 +1,80 @@
...
+          'text-accent-500',
+          'hover:text-accent-100 active:text-accent-700 focus:outline-accent-500',
+        ],
+      },
+    },
+  }
+)
 
-  const mode = primary ? primaryClasses : secondaryClasses
+export type Props = ComponentProps<'button'> &
```
[the remark this answers is not in the document]
Comment (pRomKol) [reply src/components/button/Button.tsx]: done.

File: src/components/button/Button.tsx [hunk 4]
```diff
@@ -1,60 +1,80 @@
...
+  disabled = false,
+  variant,
+  ...props
+}: Props) => {
   return (
     <button
-      className={`${baseClasses} ${sizeClasses[size]} ${mode}`}
-      style={{ backgroundColor }}
-      type='button'
+      className={twMerge(cn(buttonVariants({ disabled, variant }), className))}
```
[the remark this answers is not in the document]
Comment (pRomKol) [reply src/components/button/Button.tsx]: done

File: src/components/button/Button.tsx [hunk 5]
```diff
@@ -1,60 +1,80 @@
...
+  }
+)
 
-  const mode = primary ? primaryClasses : secondaryClasses
+export type Props = ComponentProps<'button'> &
+  VariantProps<typeof buttonVariants>
 
+export const Button = ({
+  className,
+  disabled = false,
```
[the remark this answers is not in the document]
Comment (pRomKol) [reply src/components/button/Button.tsx]: done

File: src/components/button/Button.tsx [hunk 6]
```diff
@@ -1,60 +1,80 @@
...
+    },
+    variants: {
+      disabled: {
+        false: '',
+        true: 'cursor-not-allowed',
+      },
+      variant: {
+        outlined: [
+          'bg-transparent text-accent-500 border border-accent-500',
+          ' enabled:hover:border-accent-100 enabled:hover:text-accent-100 active:text-accent-700 active:border-accent-700',
```
[the remark this answers is not in the document]
Comment (pRomKol) [reply src/components/button/Button.tsx]: done

File: src/components/button/Button.tsx [hunk 7]
```diff
@@ -1,60 +1,80 @@
...
+        ],
+        primary: [
+          'bg-accent-500',
+          ' enabled:hover:bg-accent-100 active:bg-accent-700 active: text-light-500',
```
[the remark this answers is not in the document]
Comment (pRomKol) [reply src/components/button/Button.tsx]: done

File: src/components/button/Button.tsx (hunk 1 above)
Comment (pRomKol) [reply src/components/button/Button.tsx]: done

File: src/components/button/Button.tsx (hunk 2 above)
Comment (pRomKol) [reply src/components/button/Button.tsx]: done

File: package.json [hunk 8]
```diff
@@ -39,6 +39,7 @@
     ]
   },
   "dependencies": {
+    "class-variance-authority": "^0.7.0",
```
Comment (sfgardev) [inline package.json:42]: remove ^ from version

Comment (pRomKol) [reply package.json]: done

**Review by SlavaMryasov — APPROVED** [review]

**Pull request merged**